js/private/js_image_layer.bzl

```diff
@@ -136,7 +136,15 @@ def _runfile_path(ctx, file, runfiles_dir):
 def _runfiles_dir(root, default_info):
     manifest = default_info.files_to_run.runfiles_manifest
 
-    runfiles = manifest.short_path.replace(manifest.basename, "")[:-1]
+    nobuild_runfile_links_is_set = manifest.short_path.endswith("_manifest")
+
+    if nobuild_runfile_links_is_set:
+        # When `--nobuild_runfile_links` is set, runfiles_manifest points to the manifest
+        # file sitting adjacent to the runfiles tree rather than within it.
+        runfiles = default_info.files_to_run.runfiles_manifest.short_path.replace("_manifest", "")
```

Comment:
I wonder if this is a safe assumption to make.

Reply:
I asked a question on the Bazel slack here: https://bazelbuild.slack.com/archives/CA31HN1T3/p1677196663633109

Reply:
I might need to check the Runfiles.java to make this is the case. is this blocking you currently?

Reply:
Yeah it's blocking some client work.